File: app/src/main/java/de/tum/in/tumcampus/models/News.java
package de.tum.in.tumcampus.models;

import java.util.Date;

import de.tum.in.tumcampus.auxiliary.Utils;

/**
 * News Object
 */
public class News {

	/**
	 * Date
	 */
	public final Date date;

	/**
	 * News Facebook-ID
	 */
	public final String id;

	/**
	 * Local image, e.g. /mnt/sdcard/tumcampus/news/cache/xy.jpg
	 */
	public final String image;

	/**
	 * Link Url, e.g. http://www.in.tum.de
	 */
	public final String link;

	/**
	 * Message, e.g. X released
	 */
	public final String message;

	/**
	 * New News
	 *
	 * @param id News Facebook-ID
	 * @param message Message, e.g. X released
	 * @param link Url, e.g. http://www.in.tum.de
	 * @param image Local image, e.g. /mnt/sdcard/tumcampus/news/cache/xy.jpg
	 * @param date Date
	 */
	public News(String id, String message, String link, String image, Date date) {

		this.id = id;
		this.message = message;
		this.link = link;
		this.image = image;
		this.date = date;
	}

	@Override
	public String toString() {
		return "id=" + id + " message=" + message + " link=" + link + " image="
				+ image + " date=" + Utils.getDateString(date);
	}
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/news/model/News.kt
package de.tum.`in`.tumcampusapp.component.ui.news.model

import android.content.Context
import android.content.Intent
import android.net.Uri
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.RoomWarnings
import app.tum.campus.api.GetNewsReply
import de.tum.`in`.tumcampusapp.component.ui.tufilm.KinoActivity
import de.tum.`in`.tumcampusapp.utils.Const
import de.tum.`in`.tumcampusapp.utils.DateTimeUtils
import de.tum.`in`.tumcampusapp.utils.toDateTime
import org.joda.time.DateTime

/**
 * New News
 *
 * @param id News Facebook-ID
 * @param title Title
 * @param link Url, e.g. http://www.in.tum.de
 * @param image Image url e.g. http://www.tu-film.de/img/film/poster/Fack%20ju%20Ghte.jpg
 * @param date Date
 * @param created Creation date
 */
@Entity
@SuppressWarnings(RoomWarnings.DEFAULT_CONSTRUCTOR)
data class News(
    @PrimaryKey
    var id: String = "",
    var title: String = "",
    var link: String = "",
    var src: String = "",
    var image: String = "",
    var date: DateTime = DateTime(),
    var created: DateTime = DateTime(),
    var dismissed: Int = 0
) {

    val isFilm: Boolean
        get() = src == "2"

    val isNewspread: Boolean
        get() = setOf(7, 8, 9, 13).contains(src.toInt())

    fun getIntent(context: Context): Intent? {
        return if (isFilm) {
            Intent(context, KinoActivity::class.java).apply {
                putExtra(Const.KINO_DATE, DateTimeUtils.getDateTimeString(date))
            }
        } else {
            if (link.isBlank()) null else Intent(Intent.ACTION_VIEW, Uri.parse(link))
        }
    }

    companion object {
        fun fromProto(it: GetNewsReply): List<News> {
            return it.newsList.map {
                News(
                    id = it.id.toString(),
                    title = it.title,
                    link = it.link,
                    src = it.source,
                    image = it.imageUrl,
                    date = it.date.toDateTime(),
                    created = it.created.toDateTime()
                )
            }
        }
    }
}
